include/deal.II/fe/fe_values_coupling.h

```diff
@@ -0,0 +1,557 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2023 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_fe_values_coupling_h
+#define dealii_fe_values_coupling_h
+
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/algorithms/general_data_storage.h>
+
+#include <deal.II/base/subscriptor.h>
+#include <deal.II/base/utilities.h>
+
+#include <deal.II/fe/fe_values_views.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+// Forward declaration
+#ifndef DOXYGEN
+template <int dim, int spacedim>
+class FEValuesBase;
+#endif
+
+namespace FEValuesViews
+{
+  /**
+   * A class that provides a renumbered view to a given FEValuesViews object.
+   *
+   * In general, the order of the degrees of freedom and quadrature points
+   * follows the one of the FEValues object. However, in some cases, it is
+   * convenient to group together degrees of freedom and quadrature points in a
+   * different order. This class provides a view to a given FEValuesViews
+   * object, where the degrees of freedom and quadrature points are renumbered
+   * according to the given renumbering vectors.
+   */
+  template <typename ViewType>
```

Comment:
There is only a single tutorial program (step-81) that ever names these `FEValuesViews` objects. (And I'm going to write a patch eventually to change that.) Rather, we only ever advertise using the `FEValuesExtractors` explicitly, and you happen to get a view when you use `fe_values[...]`. 

Have you considered whether one could use the extractor type as a template argument instead, to stick with things that typically appear in user programs? You could then value a
```
  using VewType = decltype(declval<FEValues<...>>()[declval<ExtractorType>()]);
```
to obtain the corresponding view type.

I would prefer something like this, because the extractor types are so much more widely used. You'd have to make `dim,spacedim` template arguments as well, though.

Reply:
That's exactly the plan. The FEValuesCoupling object will take extractors and return a `ReorderedView`, pretty much like `FEValues` does.

Reply:
Ah, I see. This isn't a class one is supposed to create objects of by hand. Then yes, this makes sense.